core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlUtils.java

```diff
@@ -117,15 +97,35 @@ protected boolean isAnalyticalRex(RexNode rexNode) {
     return false;
   }
 
-  private boolean isRexOverPresentInRexCall(RexNode rexNode) {
-    if (rexNode instanceof RexCall) {
-      List<RexNode> listOfRexNode = ((RexCall) rexNode).getOperands();
-      for (RexNode node : listOfRexNode) {
-        if (node instanceof RexOver) {
-          return true;
-        }
+  /** Walks over an expression and determines whether it is RexOver.
+   */
+  private static class AnalyticalFunFinder extends RexVisitorImpl<RexNode> {
```

Comment:
if this shuttles purpose is to just check for the existence of a RexOver then this should be <code>RexVisitorImpl\<Boolean\></code> instead of <code>RexVisitorImpl\<RexNode\></code>

Reply:
@Katakam-Phani-Mahidhar chenged the logic.Please have a look